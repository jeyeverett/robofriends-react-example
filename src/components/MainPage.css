@font-face {
    font-display: swap;
    font-family: 'Bulgatti';
    font-style: normal;
    font-weight: normal;
    src: local('Bulgatti'), url('../containers/Bulgatti.woff') format('woff');
}

h1 {
    font-family: 'Bulgatti';
    font-weight: 200;
    color: #0ccac4;
}