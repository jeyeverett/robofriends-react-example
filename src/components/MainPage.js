import React, { Component } from 'react';
import CardList from '../components/CardList'; //We imported all Cards into CardList and then imported that here
import SearchBox from '../components/SearchBox';
import Scroll from '../components/Scroll'; //Scroll is a custom component we created
import ErrorBoundary from '../components/ErrorBoundary';
import Header from '../components/Header';
import './MainPage.css';

class MainPage extends Component {

    //This is a built in React lifecycle hook that will run as soon as our component is mounted
    componentDidMount() {
        this.props.onRequestRobots();
    }

    filteredRobots = () => {
        return this.props.robots.filter((robot) => {
            return robot.name.toLowerCase().includes(this.props.searchfield.toLowerCase());
        });
    }

    render() {
        const { onSearchChange, isPending } = this.props;


        //The if statement below accounts for possibility of a delay due to a large robots array i.e. we show the 'LOADING' heading until the document is ready
        //Note that we refactored this to use the ternary operator (it is a cleaner syntax for if else statements)
        return isPending ? 
            <h1 className='tc'>LOADING</h1> :   
            (
                <div className='tc'>
                    <Header />
                    <SearchBox searchChange={onSearchChange} />
                    <Scroll>
                        <ErrorBoundary>
                            <CardList robots={this.filteredRobots()}/>
                        </ErrorBoundary>
                    </Scroll>
                </div>
            );
    }
}

export default MainPage;
