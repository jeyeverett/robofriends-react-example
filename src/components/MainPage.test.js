import { create } from "react-test-renderer";
import React from 'react';
import MainPage from './MainPage';

let main;

//Before each will run at the beginning of every test
beforeEach(() => {
    const mockProps = {
        onRequestRobots: jest.fn(),
        robots: [],
        searchfield: '',
        isPending: false
    }
    main = create(<MainPage {...mockProps} />);
});

it('expect to render MainPage component', () => {
    expect(main.toJSON()).toMatchSnapshot();
});

it('filters the robots correctly', () => {
    const mockProps2 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchfield: 'john',
        isPending: false
    }
    const main2 = create(<MainPage {...mockProps2} />);
    const instance = main2.root;

    expect(instance._fiber.stateNode.filteredRobots()).toEqual([{
        id: 3,
        name: 'John',
        email: 'john@gmail.com'
    }]);
});

it('filters the robots correctly 2', () => {
    const mockProps3 = {
        onRequestRobots: jest.fn(),
        robots: [{
            id: 3,
            name: 'John',
            email: 'john@gmail.com'
        }],
        searchfield: 'a',
        isPending: true
    }
    const main3 = create(<MainPage {...mockProps3} />);
    const instance = main3.root;
    const filteredRobots = [];

    expect(instance._fiber.stateNode.filteredRobots()).toEqual(filteredRobots);
});