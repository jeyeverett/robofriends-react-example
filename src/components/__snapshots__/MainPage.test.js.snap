// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render MainPage component 1`] = `
<div
  className="tc"
>
  <div>
    <h1
      className="f1"
    >
      RoBros
    </h1>
    <button
      id="counter"
      onClick={[Function]}
    >
      Count: 
      0
    </button>
  </div>
  <div>
    <input
      aria-label="Search Robros"
      className="pa3 mb3 br4 tc ba b--green bg-lightest-blue w5"
      placeholder="Search RoBros"
      type="search"
    />
  </div>
  <div
    style={
      Object {
        "backgroundColor": "white",
        "border": "1px solid white",
        "height": "500px",
        "overflowY": "scroll",
      }
    }
  >
    <div />
  </div>
</div>
`;
